dds/DCPS/DiscoveryBase.h

```diff
@@ -2375,7 +2378,7 @@ namespace OpenDDS {
         }
```

Comment:
Should iter == participants_.end() when this function returns?

Reply:
I think so, yeah, but I'm not sure if it's possible to reregister while disassociate's reverse lock is in effect.